slider/src/components/Map.vue

```diff
@@ -0,0 +1,159 @@
+
+export default {
+  name: 'map',
+  methods: {
+    onMoveEnd(event) {
+      const map = event.target;
+      if (map) {
+        this.center = map.getCenter();
+        const query = {
+          coordinates: `${this.center.lat},${this.center.lng}`,
+          zoom: this.zoom,
+        };
+        if (this.leftProject.length) {
+          query.left = this.leftProject;
+        }
+        if (this.rightProject.length) {
+          query.right = this.rightProject;
+        }
+        if (this.leftToken.length) {
+          query.mapTokenLeft = this.leftToken;
+        }
+        if (this.rightToken.length) {
+          query.mapTokenRight = this.rightToken;
+        }
+        this.$router.replace({
+          name: this.$router.currentRoute.name,
+          query,
+        });
+      }
+    },
+    onZoomEnd({ target: { _zoom: v } }) {
+      this.zoom = v;
+      const query = {
+        coordinates: `${this.center.lat},${this.center.lng}`,
+        zoom: v,
+      };
+      if (this.leftProject.length) {
+        query.left = this.leftProject;
+      }
+      if (this.rightProject.length) {
+        query.right = this.rightProject;
+      }
+      if (this.leftToken.length) {
+        query.mapTokenLeft = this.leftToken;
+      }
+      if (this.rightToken.length) {
+        query.mapTokenRight = this.rightToken;
+      }
+      this.$router.replace({
+        name: this.$router.currentRoute.name,
+        query,
+      });
+    },
+  },
+  beforeMount() {
+    const query = this.$route.query;
+    const zoom = parseInt(query.zoom, 10);
+    if (Number.isFinite(zoom)) {
+      this.zoom = zoom;
+    }
+    if (query.coordinates) {
+      const coords = query.coordinates.split(',');
+      if (coords.length === 2) {
+        this.center = L.latLng(coords);
+      }
+    }
+    if (query.mapTokenLeft) {
+      this.leftToken = query.mapTokenLeft;
+    }
+    if (query.mapTokenRight) {
+      this.rightToken = query.mapTokenRight;
+    }
+    if (query.left) {
+      this.leftProject = query.left;
+    }
+    if (query.right) {
+      this.rightProject = query.right;
+    }
+    if (query.mapTokenLeft) {
+      this.leftToken = query.mapTokenLeft;
+    }
+    if (query.mapTokenRight) {
+      this.rightToken = query.mapTokenRight;
+    }
+  },
+  mounted() {
+    const map = this.$refs.map.mapObject;
+    L.control.sideBySide(
+      this.$refs.leftLayer.mapObject,
+      this.$refs.rightLayer.mapObject,
+    ).addTo(map);
+  },
+  data() {
+    return {
+      dividerPosition: 0.5,
+      parentFrame: null,
+      leftFrame: null,
+      rightFrame: null,
+      zoom: 14,
+      basemapParams: { maxZoom: 30, maxNativeZoom: 18 },
+      layerParams: { maxZoom: 30 },
+      leftProject: '',
+      rightProject: '',
+      leftToken: '',
+      rightToken: '',
+      center: L.latLng([
+        39.9526,
+        -75.1652,
+      ]),
+    };
+  },
+  computed: {
+    basemapUrl() {
+      return 'http://{s}.tile.osm.org/{z}/{x}/{y}.png';
```

Comment:
Can we make this HTTPS?

Reply:
I'll probably add a component similar to the basemap selector we use in rasterfoundry.

Reply:
That would be great to add all of our basemaps, we have to whitelabel our domain probably